projects/vdk-plugins/vdk-structlog/tests/test_plugin.py

```diff
@@ -1,33 +1,81 @@
+            ["run", jobs_path_from_caller_directory("job-with-bound-logger")]
+        )
+
+        logs = result.output.split("\n")
+
+        log_with_no_bound_context = [
+            x for x in logs if "Log statement with no bound context" in x
+        ][0]
+        log_with_bound_context = [
+            x for x in logs if "Log statement with bound context" in x
+        ][0]
+        log_with_bound_and_extra_context = [
+            x for x in logs if "Log statement with bound context and extra context" in x
+        ][0]
+
+        # check for job name in logs
+        assert "job-with-bound-logger" in log_with_no_bound_context
+        assert "job-with-bound-logger" in log_with_bound_context
+        assert "job-with-bound-logger" in log_with_bound_and_extra_context
+
+        # check bound logger can bind context but that bound context does not appear in the rest of logging
+        assert "bound_test_value" in log_with_bound_context
+        assert "bound_test_value" not in log_with_no_bound_context
+
+        # check extra bound context does not appear if not set in the logging_metadata config variable
+        assert "extra_test_value" not in log_with_bound_context
+
+        # check for both the bound and the extra context in bound log statements which include extra context
+        assert "extra_test_value" in log_with_bound_and_extra_context
+        assert "bound_test_value" in log_with_bound_and_extra_context
+
+        # check that one of the bound values does not appear in the logs since we've not configured it to appear
+        assert "excluded_value" not in log_with_bound_context
+        assert "excluded_value" not in log_with_bound_and_extra_context
+
+        # check the log level and job name appear in the logs (so we can compare to when we exclude them below)
+        assert "[INFO ]" in result.output
+        assert "job-with-bound-logger" in result.output
 
-def test_dummy():
     with mock.patch.dict(
         os.environ,
         {
-            # mock the vdk configuration needed for our test
+            "VDK_LOGGING_METADATA": "timestamp,file_name,line_number,bound_test_key,extra_test_key"
         },
     ):
-        # CliEntryBasedTestRunner (provided by vdk-test-utils) gives a away to simulate vdk command
-        # and mock large parts of it - e.g passed our own plugins
         runner = CliEntryBasedTestRunner(structlog_plugin)
 
         result: Result = runner.invoke(
-            ["run", jobs_path_from_caller_directory("job-using-a-plugin")]
+            ["run", jobs_path_from_caller_directory("job-with-bound-logger")]
         )
-        cli_assert_equal(0, result)
+
+        test_log = [
+            x
+            for x in result.output.split("\n")
+            if "Log statement with bound context and extra context" in x
+        ][0]
+
+        # check that log level and job name do not appear if we have not included them in logging_metadata
+        assert "[INFO ]" not in test_log
```

Comment:
Can you also test this for each canonical metadata field, e.g. timestamp, file_name, line_number, etc.? These fields are basically hiding %-format specifiers, so it would be nice to see that each of them filter correctly.

You can maybe do this using a loop and re-setting the environment for each execution, then checking that the metadata field is missing.

Reply:
did this for all except timestamps because I need to work around with regex and it overcomplicates the test